applications/FluidDynamicsApplication/custom_utilities/qsvms_dem_coupled_data.h

```diff
@@ -73,15 +76,43 @@ void Initialize(
     this->FillFromHistoricalNodalData(FluidFraction, FLUID_FRACTION, r_geometry);
     this->FillFromHistoricalNodalData(FluidFractionRate, FLUID_FRACTION_RATE, r_geometry);
     this->FillFromHistoricalNodalData(FluidFractionGradient, FLUID_FRACTION_GRADIENT, r_geometry);
+    this->FillFromHistoricalNodalData(Permeability, PERMEABILITY, r_geometry);
     this->FillFromHistoricalNodalData(MassSource, MASS_SOURCE, r_geometry);
     this->FillFromHistoricalNodalData(Acceleration, ACCELERATION, r_geometry);
     this->FillFromHistoricalNodalData(BodyForce,BODY_FORCE,r_geometry);
 
     ElementSize = ElementSizeCalculator<TDim,TNumNodes>::MinimumElementSize(r_geometry);
 }
 
-///@}
+void FillFromHistoricalNodalData(
+    NodalTensorData& rData,
+    const Variable<Matrix>& rVariable,
+    const Geometry<Node<3>>& rGeometry)
+{
+    rData.resize(TNumNodes);
+    for (size_t i = 0; i < TNumNodes; i++) {
+        const Matrix& r_nodal_values =
+            rGeometry[i].FastGetSolutionStepValue(rVariable);
+        rData[i] = r_nodal_values;
+    }
+}
+
+void FillFromHistoricalNodalData(
+    NodalVectorData& rData,
+    const Variable<array_1d<double, 3>>& rVariable,
+    const Geometry<Node<3>>& rGeometry)
+{
+    QSVMSData<TDim, TNumNodes, TElementIntegratesInTime>::FillFromHistoricalNodalData(rData, rVariable, rGeometry);
+}
 
+void FillFromHistoricalNodalData(
+    NodalScalarData& rData,
+    const Variable<double>& rVariable,
+    const Geometry<Node<3>>& rGeometry)
+{
+    QSVMSData<TDim, TNumNodes, TElementIntegratesInTime>::FillFromHistoricalNodalData(rData, rVariable, rGeometry);
+}
```

Comment:
Why do you need to reimplement these? Aren't these equal to the base class ones?

Reply:
Yes, but compiler throws an error if I do not reimplement them.

Reply:
Strange... this mechanism is exploited throughout all the elements... Indeed, why this wasn't happening before?

Reply:
It does not matter, I have implemented it in the base class